include/secp256k1_schnorrsig.h

```diff
@@ -50,59 +52,125 @@ typedef int (*secp256k1_nonce_function_hardened)(
  *
  *  If a data pointer is passed, it is assumed to be a pointer to 32 bytes of
  *  auxiliary random data as defined in BIP-340. If the data pointer is NULL,
- *  schnorrsig_sign does not produce BIP-340 compliant signatures. The algo16
- *  argument must be non-NULL, otherwise the function will fail and return 0.
- *  The hash will be tagged with algo16 after removing all terminating null
- *  bytes. Therefore, to create BIP-340 compliant signatures, algo16 must be set
- *  to "BIP0340/nonce\0\0\0"
+ *  schnorrsig_sign does not follow the BIP-340 nonce derivation procedure
+ *  exactly. The algo16 argument must be non-NULL, otherwise the function will
+ *  fail and return 0. The hash will be tagged with algo16 after removing all
+ *  terminating null bytes. Therefore, to create BIP-340 compliant signatures,
+ *  algo16 must be set to "BIP0340/nonce\0\0\0"
```

Comment:
Maybe it's easier than having a length but I think if we have a length arg for the message, we can have it here too. `\0\0\0` is somewhat strange. I mean it works but stripping trailing \0 bytes is very unconventional, I haven't seen this anywhere else.

Reply:
added `algolen` argument